lib/dockerregistry/checks.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	gogit "gopkg.in/src-d/go-git.v4"
+)
+
+// ImageRemovalCheck implements the PreCheck interface and checks against
+// pull requests that attempt to remove any images from the promoter manifests.
+type ImageRemovalCheck struct {
+	ManifestRepo string
+	MasterSHA    string
```

Comment:
A string is fine here but since you're using gogit, isn't there some SHA type you can use? I just found https://godoc.org/gopkg.in/src-d/go-git.v4/plumbing#Hash